server/routes/termdb.categories.ts

```diff
@@ -89,43 +83,19 @@ function init({ genomes }) {
 
 async function trigger_getcategories(
 	q: {
-		tid: string | number
-		type: string
-		filter: any
+		term: Term
 		term1_q: any
-		name?: string
-		gene?: string
-		chr?: number
-		start?: number
-		stop?: number
+		filter: any
 		currentGeneNames?: string[]
 		rglst?: any
 	},
+
 	res: { send: (arg0: { lst: any[]; orderedLabels?: any }) => void },
 	tdb: { q: { termjsonByOneid: (arg0: any) => any } },
 	ds: { assayAvailability: { byDt: { [s: string]: any } | ArrayLike<any> } },
 	genome: any
 ) {
-	// thin wrapper of get_summary
-	// works for all types of terms
-	if (!q.tid) throw '.tid missing'
-	let term
-	if (q.type == 'geneVariant') {
-		term = { id: q.name, name: q.name, type: 'geneVariant', isleaf: true }
-		if (q.gene) {
-			term.gene = q.gene
-		} else {
-			term.chr = q.chr
-			term.start = q.start
-			term.stop = q.stop
-		}
-	} else {
-		term = tdb.q.termjsonByOneid(q.tid)
-	}
-	const tw =
-		q.type == 'geneVariant'
-			? { $id: q.gene || q.name || q.tid, term, q: { isAtomic: true } }
-			: { $id: q.tid, id: q.tid, term, q: q.term1_q || getDefaultQ(term, q) }
+	const tw: TermWrapper = { term: q.term, q: q.term1_q || getDefaultQ(q.term, q) }
```

Comment:
how about we still create $id with random string so later when deleting auto-assigning tw.$id from getData(), this code won't need change